src/commands/openYAMLConfigFile.ts

```diff
@@ -45,14 +47,67 @@ export async function openYAMLConfigFile(context: IActionContext, node?: StaticW
     await window.showTextDocument(configDocument, { selection });
 }
 
-async function getSelection(configDocument: TextDocument, buildConfigToSelect: BuildConfig): Promise<Range | undefined> {
-    const configRegex: RegExp = new RegExp(`${buildConfigToSelect}:`);
+/* eslint-disable @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-member-access, no-prototype-builtins, @typescript-eslint/no-unsafe-assignment */
+export async function getSelection(configDocument: TextDocument, buildConfigToSelect: BuildConfig): Promise<Range | undefined> {
+    const configDocumentText: string = configDocument.getText();
+    const buildConfigRegex: RegExp = new RegExp(`${buildConfigToSelect}:`, 'g');
+    const buildConfigMatches: RegExpMatchArray | null = configDocumentText.match(buildConfigRegex);
 
-    let offset: number = configDocument.getText().search(configRegex);
-    // Shift the offset to the beginning of the build config's value
-    offset += `${buildConfigToSelect}: `.length;
+    if (buildConfigMatches && buildConfigMatches.length > 1) {
+        void ext.ui.showWarningMessage(localize('foundMultipleBuildConfigs', 'Multiple "{0}" build configurations were found in "{1}".', buildConfigToSelect, basename(configDocument.uri.fsPath)));
+        return undefined;
+    }
+
+    const parsedYaml: Document.Parsed = parseDocument(configDocumentText);
+    type YamlNode = YAMLMap | YAMLSeq | Pair | undefined;
+    const yamlNodes: YamlNode[] = [];
+    let yamlNode: YamlNode = parsedYaml.get('jobs');
+
+    while (yamlNode) {
+        if (yamlNode.hasOwnProperty('key') && yamlNode['key'].value === buildConfigToSelect && yamlNode.hasOwnProperty('value')) {
+            const configValue = <Scalar>yamlNode['value'];
+            const range = <number[] | undefined>configValue.range;
+
+            if (range) {
+                const buildConfigOffset: number = configDocumentText.search(buildConfigRegex);
+                let newlines: number = 0;
+
+                // The range returned from the `yaml` package doesn't include newlines
+                // So count newlines and include them in the range we return
+                for (const char of configDocumentText.slice(0, buildConfigOffset)) {
+                    newlines += char === '\n' ? 1 : 0;
```

Comment:
There seems to be another way to do this without the for-loop. 
Will, you might have already looked at this, but wondering if we can use this method instead? 
https://github.com/eemeli/yaml/issues/127#issuecomment-628184035

Also, should we track this issue? In case there is an improvement from the yaml side.

Reply:
Thanks for the suggestion - now this is using only the `yaml` package to get ranges. I don't think we need to track the issue though. From the looks of it they've already rejected the feature request to include newline counts in ranges